nvflare/client/sub_process/process_api.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import os
+from typing import Any, Dict, Optional, Tuple, Union
+
+from nvflare.apis.analytix import AnalyticsDataType
+from nvflare.apis.fl_constant import FLMetaKey
+from nvflare.apis.utils.analytix_utils import create_analytic_dxo
+from nvflare.app_common.abstract.fl_model import FLModel
+from nvflare.client.api_spec import APISpec
+from nvflare.client.config import ClientConfig, ConfigKey, ExchangeFormat, from_file
+from nvflare.client.constants import CLIENT_API_CONFIG
+from nvflare.client.flare_agent import FlareAgentException
+from nvflare.client.flare_agent_with_fl_model import FlareAgentWithFLModel
+from nvflare.client.model_registry import ModelRegistry
+from nvflare.fuel.utils import fobs
+from nvflare.fuel.utils.import_utils import optional_import
+from nvflare.fuel.utils.pipe.pipe import Pipe
+
+
+def _create_client_config(config: Union[str, Dict]) -> ClientConfig:
+    if isinstance(config, str):
+        client_config = from_file(config_file=config)
+    elif isinstance(config, dict):
+        client_config = ClientConfig(config=config)
+    else:
+        raise ValueError("config should be either a string or dictionary.")
+    return client_config
```

Comment:
some of these codes changes, please refer to: https://github.com/NVIDIA/NVFlare/blob/main/nvflare/client/api.py